packages/math/src/integers.ts

```diff
@@ -238,10 +238,10 @@ export class Uint64 implements Integer, WithByteConverters {
 
   private constructor(data: BN) {
     if (data.isNeg()) {
-      throw new Error("Input is negative");
+      throw new Error("Input is negative. Found: " + data);
```

Comment:
Does that give you a proper debug representation of a BN?

Reply:
I don't think so, I will adapt this with the call toString()